src/Compilers/Core/Portable/Operations/BasicBlock.cs

```diff
@@ -92,21 +93,33 @@ public ImmutableArray<ControlFlowBranch> Predecessors
         /// </summary>
         public ControlFlowRegion EnclosingRegion { get; }
 
-        internal void SetSuccessorsAndPredecessors(ControlFlowBranch successor, ControlFlowBranch conditionalSuccessor, ImmutableArray<ControlFlowBranch> predecessors)
+        internal void SetSuccessors(ControlFlowBranch successor, ControlFlowBranch conditionalSuccessor)
         {
 #if DEBUG
-            Debug.Assert(!_sealed);
+            Debug.Assert(!_successorsAreSealed);
             Debug.Assert(_lazySuccessor == null);
             Debug.Assert(_lazyConditionalSuccessor == null);
-            Debug.Assert(_lazyPredecessors.IsDefault);
 #endif
 
             _lazySuccessor = successor;
             _lazyConditionalSuccessor = conditionalSuccessor;
+
+#if DEBUG
+            _successorsAreSealed = true;
+#endif
+        }
+
+        internal void SetPredecessors(ImmutableArray<ControlFlowBranch> predecessors)
```

Comment:
> Should we assert that the predecessors isn't default?

We didn't before. 

---
In reply to: [192519377](https://github.com/dotnet/roslyn/pull/27363#discussion_r192519377) [](ancestors = 192519377)

Reply:
But should we?

Reply:
> But should we?

There is no requirement to assert everything, I think. I do not believe it is necessary because there is really a single consumer of this API and it does the right thing. I do not expect this to change in the future

---
In reply to: [192521061](https://github.com/dotnet/roslyn/pull/27363#discussion_r192521061) [](ancestors = 192521061)